libraries/botbuilder-dialogs-adaptive/src/actions/httpRequest.ts

```diff
@@ -286,82 +287,118 @@ export class HttpRequest<O extends object = {}> extends Dialog<O> implements Htt
+
+            switch (this.responseType.getValue(dc.state)) {
+                case ResponsesTypes.Activity:
+                    result.content = await response.json();
+                    dc.context.sendActivity(result.content as Activity);
+                    break;
+                case ResponsesTypes.Activities:
+                    result.content = await response.json();
+                    dc.context.sendActivities(result.content as Activity[]);
+                    break;
+                case ResponsesTypes.Json: {
+                    const content = await response.text();
+                    try {
+                        result.content = JSON.parse(content);
+                    } catch {
+                        result.content = content;
+                    }
+                    break;
                 }
-                break;
+                case ResponsesTypes.Binary: {
+                    const buffer = await response.arrayBuffer();
+                    result.content = new Uint8Array(buffer);
+                    break;
+                }
+                case ResponsesTypes.None:
+                default:
+                    break;
             }
-            case ResponsesTypes.Binary: {
-                const buffer = await response.arrayBuffer();
-                result.content = new Uint8Array(buffer);
-                break;
+
+            traceInfo.response = result;
+
+            await this.EndDialogWithResultAsync(dc, result, traceInfo, this.id);
+        } catch (err) {
+            const result = new Result();
+            result.content = err.message;
+            traceInfo.response = result;
+
+            if (err instanceof FetchError) {
+                result.statusCode = StatusCodes.NOT_FOUND;
+            } else {
+                throw err;
             }
-            case ResponsesTypes.None:
-            default:
-                break;
+
+            await this.EndDialogWithResultAsync(dc, result, traceInfo, this.id);
         }
+    }
 
-        traceInfo.response = result;
+    /**
+     * Writes Trace Activity for the http request and response values and returns the actionResult as the result of this operation.
+     *
+     * @param dc The [DialogContext](xref:botbuilder-dialogs.DialogContext) for the current turn of conversation.
+     * @param result Optional. Value returned from the dialog that was called. The type
+     * of the value returned is dependent on the child dialog.
+     * @param traceInfo Optional. The text to be spoken by your bot on a speech-enabled channel.
+     * @param label Optional. Indicates whether your bot is accepting, expecting, or ignoring user.
+     * @returns A `Promise` representing the asynchronous operation.
+     */
+    async EndDialogWithResultAsync(
+        dc: DialogContext,
+        result: Result,
+        traceInfo: any,
+        label: string
+    ): Promise<DialogTurnResult<any>> {
+        //traceInfo.response = result;
```

Comment:
why is this comment here, why are we passing the label, that could be just used here as this.id like in dotnet?

Reply:
Done. Thanks!